core/src/main/scala/filodb.core/binaryrecord2/RecordSchema.scala

```diff
@@ -24,12 +25,15 @@ import filodb.memory.format.{vectors => bv}
  *   the fact that all variable length fields after the partitionFieldStart are contiguous and can be binary compared
  *
  * @param columnTypes In order, the field or column type of each field in this schema
+ * @param brSchema schema of any binary record type column
  * @param partitionFieldStart Some(n) from n to the last field are considered the partition key.  A field number.
  * @param predefinedKeys A list of predefined keys to save space for the tags/MapColumn field(s)
  */
-final class RecordSchema(val columnTypes: Seq[Column.ColumnType],
+final class RecordSchema(val colNames: Seq[String],
```

Comment:
IMO if you are going to have both the names and the types, it would be much simpler just to take in the columns, ie `val columns: Seq[Column]` instead of just names and types.

Reply:
I'd prefer one less parameter

Reply:
Sure. Will try this out.